src/AST/PropertyExtensions.cs

```diff
@@ -14,7 +14,7 @@ public static bool IsInRefTypeAndBackedByValueClassField(this Property p)
             type = type ?? p.Field.Type;
 
             Class decl;
-            return type.TryGetClass(out decl) && decl.IsValueType;
+            return type.TryGetClass(out decl) && (decl.IsValueType || decl.IsStruct);
```

Comment:
It seems to me that you use this || expression any time IsValueType is used. In that case why don't you move || decl.IsStruct to the getter of IsValueType analogous to what you did for IsRefType? This way we can continue checking just IsValueType and we'll also have many less changes in this pull.